src/dao/mongo/activities.ts

```diff
@@ -0,0 +1,43 @@
+import * as _ from 'lodash';
+import * as MongoDB from 'mongodb';
+
+import { LoggerFactory } from '../../logger';
+import BaseDAO, { BaseModel } from './base';
+
+// TODO: move these models out of the 'mongo dao' in case we want other sources besides mongo.
+export interface Activity extends ActivityCreateOptions, BaseModel {}
+
+export interface ActivityCreateOptions {
+  userId: string;
+  name: string;
+  type: string;
+  startDate: Date;
+  movingTime: number;
+  totalTime: number;
+  distance: number;
+  elevation: number;
+  provider: string;
+  providerId: string;
```

Comment:
I'm on the fence about duplicating this data (since it already exists on the user). Thoughts?

Reply:
This is actually more to identify which platform the activity came from (strava, ridewithgps, etc). providerId would be the id as it exists on strava. I had named it 'platform' before, but switched it to 'provider' as we could share the same enums between user/activities. But provider might be specific to oauth?